src/payment/strategies/braintree/braintree-paypal-payment-strategy.ts

```diff
@@ -123,12 +123,32 @@ export default class BraintreePaypalPaymentStrategy implements PaymentStrategy {
             throw new InvalidArgumentError('Vaulting is disabled but shouldSaveInstrument is set to true');
         }
 
+        const shippingAddress = state.shippingAddress.getShippingAddress();
+
+        let braintreeAddress: BraintreeAddress | undefined;
+
+        if (shippingAddress) {
+            braintreeAddress = {
```

Comment:
Could we move this to a function (maybe on its own file) and add some tests for it? 

```ts
function mapToBraintreeAddress(address: ShippingAddress): BraintreeAddress
```